### PR TITLE
[HDInsight] Support Private Link and Encryption In Transit feature

### PR DESCRIPTION
**Pull request by aim-for-better**
## Description
1.  Support Private Link feature
2. Support Encryption In Transit feature
3. Update related tests and help doc
4. Update ChangeLog.md

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/HDInsight/HDInsight.Test/ScenarioTests/TestController.cs [hunk 1]
```diff
@@ -44,6 +45,7 @@ public class TestController : RMTestBase
         public KeyVaultManagementClient KeyVaultManagementClient { get; private set; }
         public KeyVaultClient KeyVaultClient { get; private set; }
         public ManagedServiceIdentityClient ManagedServiceIdentityClient { get; private set; }
+        public NetworkManagementClient NetworkManagementClient;
```
Comment (sunsw1994) [inline src/HDInsight/HDInsight.Test/ScenarioTests/TestController.cs:48]:
``` 
public NetworkManagementClient NetworkManagementClient { get; private set; }
```

File: src/HDInsight/HDInsight.Test/ScenarioTests/TestHelper.cs [hunk 2]
```diff
@@ -144,5 +149,31 @@ public static Vault GetVault(string resourceGroupName, string vaultName)
...
+                    }
+                },
+                Subnets = new List<Subnet>()
+                {
+                    new Subnet()
+                    {
+                        Name = subnetName,
+                        AddressPrefix = "10.0.0.0/24",
+                        PrivateEndpointNetworkPolicies = subnetPrivateEndpointNetworkPoliciesFlag ? "Enabled" : "Disabled",
+                        PrivateLinkServiceNetworkPolicies =subnetPrivateLinkServiceNetworkPoliciesFlag ? "Enabled" : "Disabled"
```
Comment (sunsw1994) [inline src/HDInsight/HDInsight.Test/ScenarioTests/TestHelper.cs:172]: Add a whitespace after "="

File: src/HDInsight/HDInsight/ManagementCommands/NewAzureHDInsightClusterCommand.cs [hunk 3]
```diff
@@ -142,7 +142,10 @@ public AzureHDInsightConfig Config
                     EncryptionAlgorithm = EncryptionAlgorithm,
                     EncryptionKeyName = EncryptionKeyName,
                     EncryptionKeyVersion = EncryptionKeyVersion,
-                    EncryptionVaultUri = EncryptionVaultUri
+                    EncryptionVaultUri = EncryptionVaultUri,
+                    PublicNetworkAccessType=PublicNetworkAccessType,
+                    OutboundPublicNetworkAccessType=OutboundPublicNetworkAccessType,
+                    EncryptionInTransit= EncryptionInTransit
```
Comment (sunsw1994) [inline src/HDInsight/HDInsight/ManagementCommands/NewAzureHDInsightClusterCommand.cs:148]: whitespace

File: src/HDInsight/HDInsight.Test/ScenarioTests/TestHelper.cs (hunk 2 above)
Comment (aim-for-better) [reply src/HDInsight/HDInsight.Test/ScenarioTests/TestHelper.cs]: > Add a whitespace after "="

fixed

File: src/HDInsight/HDInsight/ManagementCommands/NewAzureHDInsightClusterCommand.cs (hunk 3 above)
Comment (aim-for-better) [reply src/HDInsight/HDInsight/ManagementCommands/NewAzureHDInsightClusterCommand.cs]: > whitespace

fixed

File: src/HDInsight/HDInsight.Test/ScenarioTests/TestController.cs (hunk 1 above)
Comment (aim-for-better) [reply src/HDInsight/HDInsight.Test/ScenarioTests/TestController.cs]: > ```
> public NetworkManagementClient NetworkManagementClient { get; private set; }
> ```

fixed

**Pull request merged**